File: backend/springboot-btcweb/src/main/java/com/demobtc/springbootbtc/dto/request/product/ProductRequest.java
package com.demobtc.springbootbtc.dto.request.product;

import lombok.Data;

import java.util.List;
import java.util.Set;


@Data
public class ProductRequest {
    private String name;

    private String description;

    private Double price;

    private boolean isActive;

    private Set<CategoryToProductRequest> categorySet;

    private List<IngredientToProductRequest> ingredientList;

    public ProductRequest(){}




}

File: frontend/react-btcweb/src/models/requests/ProductRequest.ts
import CategorySetRequest from "./CategorySetRequest";
import IngredientListRequest from "./IngredientListRequest";

class ProductRequest {
  name: string;
  description: string;
  price: number;
  active: boolean;
  categorySet: CategorySetRequest[];
  ingredientList: IngredientListRequest[];

  constructor(
    name: string, 
    description: string, 
    price: number, 
    active: boolean, 
    categorySet: CategorySetRequest[], 
    ingredientList: IngredientListRequest[]
    ){
    this.name = name;
    this.description = description;
    this.price = price;
    this.active = active;
    this.categorySet = categorySet;
    this.ingredientList = ingredientList;
  }
}

export default ProductRequest;